File: Practica 3/python/Cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = []
        self.inicio = 0
        self.fin = -1
        self.n = n

    def insert(self, e):
        if self.isFull():
            print("Cola llena: no se puede insertar")
        else:
            self.fin += 1
            self.arreglo.append(e)
            print("Elemento añadido:", e)

    def remove(self):
        if self.isEmpty():
            print("Cola vacía: no se puede eliminar")
        else:
            e = self.arreglo[self.inicio]
            self.inicio += 1
            if self.isEmpty():
                self.arreglo = []
                self.inicio = 0
                self.fin = -1
            return e

    def peek(self):
        if self.isEmpty():
            print("Cola vacía: no se puede hacer peek")
        else:
            return self.arreglo[self.inicio]

    def isEmpty(self):
        return self.size() == 0

    def isFull(self):
        return self.size() == self.n

    def size(self):
        return self.fin - self.inicio + 1

cola = Cola(3)

cola.insert(10)
cola.insert(20)
cola.insert(30)

cola.insert(40)  

print("Primer elemento de la cola:", cola.peek())  

print("Elemento eliminado:", cola.remove())  
print("Elemento eliminado:", cola.remove())

print("Primer elemento de la cola:", cola.peek()) 

print("Elemento eliminado:", cola.remove())  

cola.remove()

File: Practica 3/java/Cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;

    public Cola(int n) {
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.n = n;
    }

    public void insert(long e) {
        if (isFull()) {
            System.out.println("Cola llena: no se puede insertar");
        } else {
            fin++;
            arreglo[fin] = e;
            System.out.println("Elemento añadido: " + e);
        }
    }

    public long remove() {
        if (isEmpty()) {
            System.out.println("Cola vacía: no se puede eliminar");
            return -1;
        } else {
            long e = arreglo[inicio];
            inicio++;
            if (isEmpty()) {
                inicio = 0;
                fin = -1;
            }
            System.out.println("Elemento eliminado: " + e);
            return e;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Cola vacía: no se puede hacer peek");
            return -1;
        } else {
            return arreglo[inicio];
        }
    }

    public boolean isEmpty() {
        return size() == 0;
    }

    public boolean isFull() {
        return size() == n;
    }

    public int size() {
        return fin - inicio + 1;
    }

    public static void main(String[] args) {
        Cola cola = new Cola(3);

        cola.insert(10);
        cola.insert(20);
        cola.insert(30);
        cola.insert(40);
        System.out.println("Primer elemento de la cola: " + cola.peek());

        cola.remove();
        cola.remove();
        cola.remove();
        cola.remove();


    }
}